webapp/packages/plugin-connections/src/ConnectionShield.tsx

```diff
@@ -23,10 +23,11 @@ export const ConnectionShield = observer<PropsWithChildren<IConnectionShieldProp
   const notificationService = useService(NotificationService);
 
   const connection = useResource(ConnectionShield, ConnectionInfoResource, connectionKey);
-  const connecting = getComputed(() => connectionKey && connection.resource.isConnecting(connectionKey));
+  const connecting = getComputed(() => (connectionKey && connection.resource.isConnecting(connectionKey)) || connection.loading);
+  const isConnectionReady = getComputed(() => !connecting && connection.data?.connected && connection.loaded && !connection.outdated);
```

Comment:
Seems like we should implement some general solution for checking whether resource entity is fully available or not. Not now, but maybe in a tech sprint